desci-models/src/trees/treeTools.ts

```diff
@@ -0,0 +1,291 @@
+import {
+  ResearchObjectComponentSubtypes,
+  ResearchObjectComponentType,
+  ResearchObjectV1,
+  ResearchObjectV1Component,
+} from "../ResearchObject";
+import {
+  AccessStatus,
+  DriveMetadata,
+  DriveObject,
+  DrivePath,
+  FileDir,
+  FileType,
+  RecursiveLsResult,
+  VirtualDriveArgs,
+} from "./treeTypes";
+
+export const DRIVE_NODE_ROOT_PATH = "root";
+
+export function fillIpfsTree(manifest: ResearchObjectV1, ipfsTree: FileDir[]) {
+  const pathToCompMap = generatePathCompMap(manifest);
+  const pathToDriveMap = generateFlatPathDriveMap(ipfsTree as DriveObject[]);
+  const pathToSizeMap = generatePathSizeMap(pathToDriveMap); //Sources dir sizes
+
+  const driveObjectTree = convertIpfsTreeToDriveObjectTree(
+    ipfsTree as DriveObject[],
+    pathToCompMap,
+    pathToSizeMap
+  );
+  // eslint-disable-next-line no-array-reduce/no-reduce
+  // const rootSize = driveObjectTree.reduce((acc, curr) => acc + curr.size, 0);
+  // const treeRoot = createVirtualDrive({
+  //   name: "Node Root",
+  //   componentType: ResearchObjectComponentType.DATA_BUCKET,
+  //   path: DRIVE_NODE_ROOT_PATH,
+  //   contains: driveObjectTree,
+  //   size: rootSize,
+  // });
+
+  // return [treeRoot];
+  return driveObjectTree;
+}
+
+export function getAncestorComponent(
+  drive: DriveObject,
+  pathToCompMap: Record<DrivePath, ResearchObjectV1Component>
+): ResearchObjectV1Component | null {
+  const pathSplit = drive.path!.split("/");
+  if (pathSplit.length < 3) return null;
```

Comment:
Any reason not to make `path` required in `DriveObject`? If so, might make sense to check it as well :)